main.py

```diff
@@ -0,0 +1,117 @@
+from datetime import datetime
+"""
+open all 3 files
+parse such that we have the log in a list
+count how many lines are in log, level=error count, level=info
+
+TIME FORMAT : 08:56:35.354
+
+top 5 most requested HTTP paths/endpoints
+"""
+
+class Logging:
+    def __init__(self):
+        self.log = []
+        self.date_time = []
+
+
+    def logs(self, file1, file2, file3):
+        file_1 = open(file1, 'r')
+        file_2 = open(file2, 'r')
+        file_3 = open(file3, 'r')
+
+        for line in file_1:
+            self.log.append(line)
+        for line in file_2:
+            self.log.append(line)
+        for line in file_3:
+            self.log.append(line)
+
+        file_1.close()
+        file_2.close()
+        file_3.close()
+
+    def get_entry_count(self):
+        return len(self.log)
+
+    def get_level_count(self, level):
+        count = 0
+
+        for line in self.log:
+            if level in line:
+                count += 1
+
+        return count
+
+    def log_times(self):
+        #TESTCASE: self.date_time.append(datetime.strptime(self.log[0].split("\"", 2)[1], '%Y-%m-%d %H:%M:%S.%f'))
+
+        #DEBUG: for count, val in enumerate(self.log):
+            #print(count)
+            #print(val.split("\"", 2)[1])
+
+        for i in range(len(self.log)):
+            if self.log[0].split("\"", 2)[0] in self.log[i].split("\"", 2)[0]:
+                self.date_time.append(datetime.strptime(self.log[i].split("\"", 2)[1], '%Y-%m-%d %H:%M:%S.%f'))
+
+    def get_popular_hour(self):
+        self.log_times()
```

Comment:
Not a big deal in an interview for your level again. But be prepared for the interviewer to ask you what happens if this log file is super large. In that case you wouldn't want to keep parsing over and over again each time the user calls a function.